app/uk/gov/hmrc/taxhistory/services/EmploymentHistoryService.scala

```diff
@@ -235,18 +243,8 @@ class EmploymentHistoryService @Inject()(
   def retrieveNpsIabds(nino: Nino, taxYear: TaxYear)(implicit hc: HeaderCarrier): Future[List[Iabd]] =
     npsConnector.getIabds(nino, taxYear.currentYear)
 
-  /**
-    * If the tax year requested is not the previous tax year, return None.
-    * TODO is this the correct behaviour?
-    */
-  def getNpsTaxAccount(nino: Nino, taxYear: TaxYear)(implicit hc: HeaderCarrier): Future[Option[NpsTaxAccount]] = {
-
-    if (taxYear.startYear != TaxYear.current.previous.startYear) {
-      Future.successful(None)
-    } else {
+  def getNpsTaxAccount(nino: Nino, taxYear: TaxYear)(implicit hc: HeaderCarrier): Future[Option[NpsTaxAccount]] =
       npsConnector.getTaxAccount(nino, taxYear.currentYear).map(Some(_))
```

Comment:
Happy to leave this for now but not sure about the approach being taken here. It seems a None case is used in a for comprehension further up this file (GitHub doesn't seem to want to let me comment on the line) and this None case is passed as part of a recover.

I'd instead expect this behaviour in the connector method itself or in this getNpsTaxAccount method rather than within the for comprehension. What do you think?

Reply:
Refactored the for comprehension to move the failure recovery down to the individual retrieve methods.